Sources/PCMessageSubscription.swift

```diff
@@ -0,0 +1,73 @@
+import Foundation
+import PusherPlatform
+
+public final class PCMessageSubscription {
+    public var delegate: PCRoomDelegate?
+    let resumableSubscription: PPResumableSubscription
+    public var logger: PPLogger
+    let basicMessageEnricher: PCBasicMessageEnricher
+
+    init(
+        delegate: PCRoomDelegate? = nil,
+        resumableSubscription: PPResumableSubscription,
+        logger: PPLogger,
+        basicMessageEnricher: PCBasicMessageEnricher
+    ) {
+        self.delegate = delegate
+        self.resumableSubscription = resumableSubscription
+        self.logger = logger
+        self.basicMessageEnricher = basicMessageEnricher
+    }
+
+    func handleEvent(eventId _: String, headers _: [String: String], data: Any) {
+        guard let json = data as? [String: Any] else {
+            self.logger.log("Failed to cast JSON object to Dictionary: \(data)", logLevel: .debug)
+            return
+        }
+
+        guard let eventTypeName = json["event_name"] as? String else {
+            self.logger.log("Event type name missing from room subscription event: \(json)", logLevel: .debug)
+            return
+        }
+
+        let expectedEventTypeName = "new_message"
+
+        guard eventTypeName == expectedEventTypeName else {
+            self.logger.log("Expected event type name to be \(expectedEventTypeName) but got \(eventTypeName)", logLevel: .debug)
+            return
+        }
+
+        guard let messagePayload = json["data"] as? [String: Any] else {
+            self.logger.log("Missing data for room subscription event: \(json)", logLevel: .debug)
+            return
+        }
```

Comment:
The returns on these methods seem to lose potential errors. Should these be reported to the presentation layer?

Reply:
This also applies to other logger side effects in here

Reply:
It's an interesting one. I come down on the no side because the expectation is that events received from the server should always have all of this information; the error case should be extremely, extremely rare. If they don't then that's an indication that something has gone wrong in an unexpected way. If that's the case, then we're putting the burden of determining whether or not to communicate something with the end user, which is not going to be obvious as it's not something that the user has done that would have led to an error of some kind.